src/fetchGraphQL.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Helpful Resources
+ * https://github.com/sibelius/create-react-app-relay-modern/blob/master/src/relay/fetchQuery.js
+ * https://github.com/relay-tools/relay-compiler-language-typescript/blob/master/example/ts/app.tsx
+ */
+
+import { SupportedChainId } from 'constants/chains'
+import { Variables } from 'react-relay'
+import { GraphQLResponse, ObservableFromValue, RequestParameters } from 'relay-runtime'
+
+import store, { AppState } from './state/index'
+
+const CHAIN_SUBGRAPH_URL: Record<number, string> = {
+  [SupportedChainId.MAINNET]: 'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3',
+  [SupportedChainId.RINKEBY]: 'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3',
+
+  [SupportedChainId.ARBITRUM_ONE]: 'https://api.thegraph.com/subgraphs/name/ianlapham/arbitrum-minimal',
+
+  [SupportedChainId.OPTIMISM]: 'https://api.thegraph.com/subgraphs/name/ianlapham/optimism-post-regenesis',
+
+  [SupportedChainId.POLYGON]: 'https://api.thegraph.com/subgraphs/name/ianlapham/uniswap-v3-polygon',
+
+  [SupportedChainId.CELO]: 'https://api.thegraph.com/subgraphs/name/jesse-sawa/uniswap-celo',
+}
+
+const headers = {
+  Accept: 'application/json',
+  'Content-type': 'application/json',
+}
+
+// Define a function that fetches the results of a request (query/mutation/etc)
+// and returns its results as a Promise:
+const fetchQuery = (params: RequestParameters, variables: Variables): ObservableFromValue<GraphQLResponse> => {
+  // TODO: figure out why this returns null
+  const chainId = (store.getState() as AppState).application.chainId
+
+  const subgraphUrl = CHAIN_SUBGRAPH_URL[chainId ?? SupportedChainId.MAINNET]
```

Comment:
what happens when chainid is a value that The Graph doesn't support? (i.e., optimistic kovan?)

Reply:
When on an unsupported chain, the pools page will not render inputs that lead to the API being called. This functions the same way the redux data store's calls functioned before, but I'll add an additional check to the function